tests/test_examples.py

```diff
@@ -1301,3 +1305,42 @@ def test_example_get_current(self):
                 -1.36284423e07,
             ],
         )
+
+
+@pytest.mark.unit
+def test_single_coil_optimization():
+    """Test that single coil (not coilset) optimization works."""
+    # testing that the objectives work and that the optimization framework
+    # works when a single coil is passed in.
+
+    opt = Optimizer("fmintr")
+    coil = FourierRZCoil()
+    coil.change_resolution(N=1)
```

Comment:
`FourierRZCurve` is non-planar, so there are multiple shapes that have constant curvature, so I don't think there is an "expected" value for any of the coefficients.

Reply:
The coil is guaranteed to be planar since we are only allowing it to have a single Fourier mode. And in this test the `coil.Z_n` stays exactly at 0 and `coil.R_n` changes from `[10, 0]` to approximately `[9, 0]` (see picture: blue is the original coil, red is optimized)

![image](https://github.com/PlasmaControl/DESC/assets/33005725/17a179bb-9d3a-4399-b287-cc6555b8ecb0)